### ISSUE TITLE
Crash while using Airport build tool

### ISSUE THREAD

**Issue by NEJulnes** [OpenLoco/OpenLoco#2098]
**Version information:**
 - Platform: Windows 10
 - Version: 23.08
 - Build/commit: d6cfed6

**Describe the bug**
When I open the airport build menu and move the build area around a slope, the game then crashes.

**To Reproduce**
1. Go to Airport build menu.
2. Click on on the Airport build menu.
3. Move your Airport build area around a slope
4. After a while it crashes.

**Expected behaviour**
The game not to crash moving the airport build area around a slope.

**Screenshots & Crash Dumps**
[openloco 2023-08-23--1311 crashdumps.zip](https://github.com/OpenLoco/OpenLoco/files/12423060/openloco.2023-08-23--1311.crashdumps.zip)

**Comment by AaronVanGeffen** [comment]
I'm having trouble reproducing this, myself. Hopefully, one of the devs that use Windows can see what's going wrong using the dumpfile you've provided, though.

**Comment by duncanspumpkin** [comment]
I've encountered this issue before but never been able to reliably reproduce it.

**Comment by NEJulnes** [comment]
The three crashes I got were from my experience fairly reproducible.

From the screen capture, could it be when its next to an industry?

**Comment by duncanspumpkin** [comment]
It's possible but it's crashing in vanilla code so it's a little hard to work out.

**Comment by duncanspumpkin** [comment]
![image](https://github.com/OpenLoco/OpenLoco/assets/1277401/a46a4613-96da-4f24-8e51-1a680117372b)
This is my easy to reproduce crash location. Its not near the industry either.

**Comment by duncanspumpkin** [comment]
Ahh, there is a track station underneath here. I think its a crash when its above a station.

**Comment by duncanspumpkin** [comment]
Yeah i can reproduce it with a very minimal example of a station on boulder breakers. 
![image](https://github.com/OpenLoco/OpenLoco/assets/1277401/25be4b78-62d9-4723-93dd-63ecbe5b68e9)

**Comment by NEJulnes** [comment]
![image](https://github.com/OpenLoco/OpenLoco/assets/123035106/9b8307cf-0b67-49f0-911a-1ca905bca6e4)

I did it again, specifically around a station, and got it to crash.

Could it be with the land adjustment? Multi-transport stations seem to work.

**Comment by duncanspumpkin** [comment]
I think this is a vanilla bug. I can reproduce this in openloco as old as v0.1.

**Comment by duncanspumpkin** [comment]
![image](https://github.com/OpenLoco/OpenLoco/assets/1277401/258cf4e0-c455-48a5-8854-c3bb76b792a9)

**Comment by Lev135** [comment]
> Yeah i can reproduce it with a very minimal example of a station on boulder breakers.

In fact, you don't need a station. It crashes with only a tunnel too:
![image](https://github.com/OpenLoco/OpenLoco/assets/35816057/780eb91c-432e-40b5-8a28-34ac578d0a8a)

**Comment by LeftofZen** [comment]
Am I missing something. I have no issue in 24.01.1. I know the track placement code was actually very recently reverse-engineered, so maybe this is fixed now?

![image](https://github.com/OpenLoco/OpenLoco/assets/7483209/d1fa9db7-0de7-4c36-ba7c-8a2107ebff22)

**Comment by LeftofZen** [comment]
Nevermind. You do need the station:
![image](https://github.com/OpenLoco/OpenLoco/assets/7483209/3bdcd53c-3774-4902-af49-e7447727c604)

**Comment by Lev135** [comment]
I'm sorry, you are definitely right. In both cases when I had the trouble airport was on the track with a station, but I somehow missed this

**Comment by duncanspumpkin** [comment]
This has now been fixed as of https://github.com/OpenLoco/OpenLoco/pull/2454

**Comment by shusaura85** [comment]
This appears to be safe to close. Seems like it was forgotten open
